File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/service/MockAuthService.java
package com.yahyeet.boardbook.model.mock.service;

import com.yahyeet.boardbook.model.entity.Match;
import com.yahyeet.boardbook.model.entity.User;
import com.yahyeet.boardbook.model.service.IAuthService;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionException;

public class MockAuthService implements IAuthService {

    private List<AuthenticationUser> mockUserDb;

    public MockAuthService(List<AuthenticationUser> db){
        this.mockUserDb = db;
    }

    @Override
    public CompletableFuture<User> login(String email, String password) {
        return CompletableFuture.supplyAsync(() -> {
            for(AuthenticationUser mockUser : mockUserDb){
                if(mockUser.email.equals(email) && mockUser.password.equals(password)){
                    return mockUser.user;
                }
            }

            throw new CompletionException(new Exception("User not found"));
        });
    }

    @Override
    public CompletableFuture<Void> logout() {
        return CompletableFuture.completedFuture(null);
    }

    @Override
    public CompletableFuture<User> signup(String email, String password, String name) {
        return CompletableFuture.supplyAsync(() -> {
            User user = new User(name);
            AuthenticationUser authUser = new AuthenticationUser(email, password, user);
            mockUserDb.add(authUser);

            return user;
        });

    }

    public class AuthenticationUser {
        public String email;
        public String password;
        public User user;
        public List<User> friends;
        public List<Match> matches;

        public AuthenticationUser(String email, String password, User user) {
            this.email = email;
            this.password = password;
            this.user = user;

            friends = new ArrayList<>();
            matches = new ArrayList<>();
        }

        public List<User> getFriends() {
            return friends;
        }

        public void setFriends(List<User> friends) {
            this.friends = friends;
        }

        public List<Match> getMatches() {
            return matches;
        }

        public void setMatches(List<Match> matches) {
            this.matches = matches;
        }
    }
}

File: application/app/src/main/java/com/yahyeet/boardbook/model/mock/service/MockAuthService.kt
package com.yahyeet.boardbook.model.mock.service

import com.yahyeet.boardbook.model.entity.Match
import com.yahyeet.boardbook.model.entity.User
import com.yahyeet.boardbook.model.service.IAuthService

import java.util.ArrayList
import java.util.concurrent.CompletableFuture
import java.util.concurrent.CompletionException

class MockAuthService(private val mockUserDb: MutableList<AuthenticationUser>) : IAuthService {

    override fun login(email: String, password: String): CompletableFuture<User> {
        return CompletableFuture.supplyAsync {
            for (mockUser in mockUserDb) {
                if (mockUser.email == email && mockUser.password == password) {
                    return@CompletableFuture.supplyAsync mockUser . user
                }
            }

            throw CompletionException(Exception("User not found"))
        }
    }

    override fun logout(): CompletableFuture<Void> {
        return CompletableFuture.completedFuture(null)
    }

    override fun signup(email: String, password: String, name: String): CompletableFuture<User> {
        return CompletableFuture.supplyAsync {
            val user = User(name)
            val authUser = AuthenticationUser(email, password, user)
            mockUserDb.add(authUser)

            user
        }

    }

    class AuthenticationUser(var email: String, var password: String, var user: User) {

        val friends: List<User>
            get() = user.friends

        val matches: List<Match>
            get() = user.matches
    }
}
